### DISCUSSION TITLE
Seeking Ways to Achieve Decentralized Swarm Flight with Crazyflie Drones and Collision Avoidance

### DISCUSSION THREAD

**Discussion by Daneto96** [bitcraze/discussions#1310] [Q&A] [answered]
Dear Community,

I am facing the challenge of realizing decentralized swarm flight with Crazyflie drones while ensuring collision avoidance among them. A crucial aspect of my project is for the drones to exchange their positions with each other and perform evasive maneuvers when a certain minimum distance is breached. I would greatly appreciate your help and advice on this matter.

While I am already familiar with the basics of drone control and programming, implementing a system that facilitates position data exchange and collision avoidance among drones is still uncharted territory for me.

I would like to tap into your experience and expertise and ask a few questions:

Which algorithms or techniques are best suited for the decentralized control of swarm drones with position data exchange and collision avoidance?
Are there any open-source projects or libraries specifically developed for this purpose that I could leverage?
What hardware or software requirements are necessary to achieve decentralized swarm flight with collision avoidance using Crazyflie drones?
Are there already any projects that tried implementing decentralized swarm behaviour?
Are there any relevant research papers or literature that could aid me in my project?
I am grateful for any kind of assistance, advice, or resources you can provide.
Thank you in advance for your support!

Best regards,
Daniel

**Comment by gemenerik** [comment, marked as answer]
Hi Daniel. While your questions are broad I can give you some initial pointers.

> What hardware or software requirements are necessary to achieve decentralized swarm flight with collision avoidance using Crazyflie drones?
> Are there already any projects that tried implementing decentralized swarm behaviour?

Check out [this blogpost on our IROS 2022 demo](https://www.bitcraze.io/2022/11/the-iros-2022-demo/) for an example of decentralized swarming and collision avoidance. This uses Crazyflies with Qi charging decks and the Lighthouse positioning system. The blogpost also links to the firmware that runs on the drones. This is just one way of doing it, of course.

> Are there any open-source projects or libraries specifically developed for this purpose that I could leverage?

While our IROS 2022 demo is based on the regular Crazyflie software stack, you can take a look at [Crazyswarm2](https://github.com/IMRCLab/crazyswarm2)

> Are there any relevant research papers or literature that could aid me in my project?

Absolutely! We have a (non-exhaustive) [list of Crazyflie related research papers ](https://www.bitcraze.io/portals/research/), there are several swarming-related papers. 

Hope that helps you get started! Don't hesitate if you have any more questions.

**Comment by Daneto96** [comment]
Thank you very much for the help!

**Comment by SuperHua888** [comment]
Hi, my research is also use this part of python code about obstacle avoiding and location sharing for the crazyswarm.
I hope we can keep comunication in this part, and i'll also update in the  community.
